### PR TITLE
Babel preset: Enable support for async generator functions

### PR DESCRIPTION
**Pull request by gziolo**
Async generator functions were enabled in Gutenberg to use with `data` module. See:
https://github.com/WordPress/gutenberg/blob/master/package.json#L108

Given that plugins might want to use the same syntax, let's enable this Babel plugin for everyone.

### CODE REVIEW

File: packages/babel-preset-default/test/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+describe( 'Babel preset default', () => {
```
[the remark this answers is not in the document]
Comment (gziolo) [reply packages/babel-preset-default/test/index.js]: Aren't we using LTS (node 8) on Travis?

[the remark this answers is not in the document]
Comment (aduth) [reply packages/babel-preset-default/test/index.js]: Only until October: https://github.com/nodejs/Release

[the remark this answers is not in the document]
Comment (gziolo) [reply packages/babel-preset-default/test/index.js]: They are much faster this time 👍 

We can simply remove this test if it isn't useful in the long run. What would you recommend instead?

[the remark this answers is not in the document]
Comment (aduth) [reply packages/babel-preset-default/test/index.js]: I don't know that I would test a preset, which is effectively intended as a configuration. Maybe a snapshot of transpiled code? That the configuration includes expected plugins?

**Pull request merged**